### PR TITLE
[5.x] Contact articles new layout with image

### PR DESCRIPTION
**Pull request by drmenzelit**
New layout for articles in contact.

### Summary of Changes
Added two new parameters in the contact component to display the intro image and the introtext of an article inside a contact.

### Testing Instructions
Create an article.
Create a contact connected with your user.
Activate the Content - Contact Plugin with redirection to Internal contact page.
Create a menu item for the article, display the author as link.
Create a menu item for contacts.
Configure the contact to display articles.

### Actual result BEFORE applying this Pull Request
A simple list of linked articles titles.

2023-07-10 09_58_54-Contact – Mozilla Firefox

### Expected result AFTER applying this Pull Request
Apply the PR, run "npm run build:css"

2023-07-10 09_59_17-Contacts_ Options - Joomla 5 - The future - Administration – Mozilla Firefox

2023-07-10 10_00_41-Contact – Mozilla Firefox

2023-07-10 14_43_55-Contacts_ Options - Joomla 5 - The future - Administration – Mozilla Firefox

2023-07-10 14_44_16-Contact – Mozilla Firefox

2023-07-10 15_38_19-Contact – Mozilla Firefox

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [X] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [X] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: components/com_contact/tmpl/contact/default_articles.php [hunk 1]
```diff
@@ -11,18 +11,51 @@
...
+    <?php if ($this->params->get('show_articles_introimage') || $this->params->get('show_articles_introtext')) :
+        foreach ($this->item->articles as $article) :
+            if ($this->params->get('show_articles_introimage')) :
+                $images     = json_decode($article->images);
+                $layoutAttr = [
+                    'src' => $images->image_intro,
+                    'alt' => empty($images->image_intro_alt) && empty($images->image_intro_alt_empty) ? false : $images->image_intro_alt,
+                ];
+            endif;
+        ?>
```
Comment (richard67) [inline components/com_contact/tmpl/contact/default_articles.php]:
```suggestion
            ?>
```
Don't ask me why, but the code style checker complains about this and suggests the change.

But maybe it would be even better to use not such a big PHP block but smaller one, e.g. one for each line.

Comment (drmenzelit) [reply components/com_contact/tmpl/contact/default_articles.php]: I changed the code

File: administrator/components/com_contact/config.xml [hunk 2]
```diff
@@ -291,6 +291,30 @@
...
+			>
+			<option value="0">JHIDE</option>
+			<option value="1">JSHOW</option>
+		</field>
+
+		<field
+			name="show_articles_introtext"
+			type="radio"
+			layout="joomla.form.field.radio.switcher"
+			label="COM_CONTACT_FIELD_ARTICLES_INTROTEXT_SHOW_LABEL"
```
Comment (Quy) [inline administrator/components/com_contact/config.xml]:
```suggestion
			name="show_articles_introimage"
			type="radio"
			label="COM_CONTACT_FIELD_ARTICLES_INTROIMAGE_SHOW_LABEL"
			layout="joomla.form.field.radio.switcher"
			default="0"
			showon="show_articles:1"
			>
			<option value="0">JHIDE</option>
			<option value="1">JSHOW</option>
		</field>

		<field
			name="show_articles_introtext"
			type="radio"
			label="COM_CONTACT_FIELD_ARTICLES_INTROTEXT_SHOW_LABEL"
			layout="joomla.form.field.radio.switcher"
```

File: administrator/language/en-GB/com_contact.ini [hunk 3]
```diff
@@ -24,6 +24,8 @@ COM_CONTACT_ERROR_UNIQUE_ALIAS="Another Contact in this category has the same al
 COM_CONTACT_ERROR_UNIQUE_ALIAS_TRASHED="A trashed Contact in this category has the same alias."
 COM_CONTACT_FIELD_ARTICLES_DISPLAY_NUM_LABEL="# Articles to List"
 COM_CONTACT_FIELD_ARTICLES_SHOW_LABEL="User Articles"
+COM_CONTACT_FIELD_ARTICLES_INTROIMAGE_SHOW_LABEL="User Articles Intro Image"
+COM_CONTACT_FIELD_ARTICLES_INTROTEXT_SHOW_LABEL="User Articles Intro Text"
```
Comment (Quy) [inline administrator/language/en-GB/com_contact.ini]:
```suggestion
COM_CONTACT_FIELD_ARTICLES_INTROIMAGE_SHOW_LABEL="User Articles Intro Image"
COM_CONTACT_FIELD_ARTICLES_INTROTEXT_SHOW_LABEL="User Articles Intro Text"
COM_CONTACT_FIELD_ARTICLES_SHOW_LABEL="User Articles"
```

Comment (Quy) [reply administrator/language/en-GB/com_contact.ini]: or change the keys?
`COM_CONTACT_FIELD_ARTICLES_SHOW_INTROIMAGE_LABEL`
`COM_CONTACT_FIELD_ARTICLES_SHOW_INTROTEXT_LABEL`

**Comment by ceford** [comment]
There seems to be no way to override the Global Intro Image and Intro text settings for an individual Contact (other than via a menu item).

With User Articles, Intro Image and Intro Text all set to Yes I get this error, seems to be for articles without Intro images:

```
Warning
: Attempt to read property "image_intro" on null in
/Users/ceford/Sites/joomla-cms/components/com_contact/tmpl/contact/default_articles.php
on line
26
```

I think work is needed!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41132">issues.joomla.org/tracker/joomla-cms/41132</a>.</sub>

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 5.1-dev.

**Comment by chmst** [comment]
Works for me. But why did you use the showon="show_articles:1" only in com.config and not for menu Items?

**Comment by richard67** [comment]
@drmenzelit It seems you've messed this PR which now has base branch 5.1-dev by upmerging branch 5.0-dev into it.

**Comment by drmenzelit** [comment]
@richard67 I have no idea how this happened ...

**Comment by drmenzelit** [comment]
@ceford can you please check if the warning dissapeared?

@chmst it seems `showon` doesn't work with fields using global options

**Comment by ceford** [comment]
I have tested this item :white_check_mark: successfully on 355fcdb3d816fc069f772a86f00e378af9443d24<br /><br />Tricky to follow the instructions but I got there - a list of articles with or without Intro Image and/or Intro Text.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41132">issues.joomla.org/tracker/joomla-cms/41132</a>.</sub>

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 355fcdb3d816fc069f772a86f00e378af9443d24<br /><br />Thanks @drmenzelit <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41132">issues.joomla.org/tracker/joomla-cms/41132</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41132">issues.joomla.org/tracker/joomla-cms/41132</a>.</sub>

**Comment by brianteeman** [comment]
The more I look at this the more I don't believe this should be in core. This PR just creates a very specific display of an article intro image and text. Which is fine but we know that people will want more control than that. 

You only have to look at the display options in the content modules to see that. 

image

Or then you will have the layout options that people might want as this looks like a blog layout but without the controls

image

Or some of the more basic options that we had to add to the Article Newsflash Module

image


In summary there is nothing in this PR that can not be achieved currently with a template override

**Comment by agimusweb** [comment]
I have tested this item :white_check_mark: successfully on 355fcdb3d816fc069f772a86f00e378af9443d24<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41132">issues.joomla.org/tracker/joomla-cms/41132</a>.</sub>

**Comment by bembelimen** [comment]
Hey @drmenzelit ,

thank you for your idea, I see your potential use case.

After many discussions I had to take a decision and I tend to agree with the statement, that this behaviour can be done via an easy override. So I will close it for now.

**Comment by fgsw** [comment]
> that this behaviour can be done via an easy override

… if you are a developer.

Not, if you are a user who can't change source codes. Overrides lacks a User Interface.
